### PR TITLE
Build and upload binary wheels in CI

### PR DESCRIPTION
**Pull request by zariiii9003**
- move project configuration from setup.py to pyproject.toml
- set minimum supported python version to 3.7
- move source code to `src` folder
- use `cibuildwheel` to build binary wheels
- remove `version.py`

### CODE REVIEW

**Comment by eerimoq** [comment]
Looks pretty good!

**Comment by eerimoq** [comment]
Building wheels is super slow. I think it's better to only run them on tagged commits and possibly when manually triggering the workflow.

**Comment by zariiii9003** [comment]
> Building wheels is super slow. I think it's better to only run them on tagged commits and possibly when manually triggering the workflow.

That's true, but it does provide some value: For every change the binaries are built and tested on every platform. 
But i can change it if you like.

**Comment by eerimoq** [comment]
I prefer fast CI, so please change it to run on tags and workflow_dispatch.

**Comment by zariiii9003** [comment]
@eerimoq I updated the workflows: the wheels will be built and uploaded when you publish a new release on GitHub.
I also run the tests through pytest for prettier output:
image

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+name: Build & Release
+
+on:
+  release:
+    types: [ published ]
```
Comment (eerimoq) [inline .github/workflows/release.yml:5]: What does this mean? How do I trigger this workflow? I usually want to publish to PyPI on all tags.

Comment (zariiii9003) [reply .github/workflows/release.yml:5]: I misread what you said earlier to be honest. So this workflow would be triggered whenever you publish a new release here on GitHub. But i just noticed, that you don't do that 😄 
image

I'll change that back so it is triggered by tags.

Comment (zariiii9003) [reply .github/workflows/release.yml]: I used the solution from [here](https://github.com/orgs/community/discussions/25302#discussioncomment-3247370)
Hopefully that works.

Comment (eerimoq) [reply .github/workflows/release.yml]: Let's try it out. I'll merge!

**Pull request merged**